### DISCUSSION TITLE
How to pass enum to options in select, please?

### DISCUSSION THREAD

**Discussion by trinketbox** [filamentphp/filament#1746] [Help]
Hi

https://github.com/laravel-filament/filament/pull/1400

From above, it looks like such a thing exists, but I can't for the life of me pass a backed php enum to a select. Any pointers please?

`Select::make('frequency')->options([frequency::cases()])`

`Select::make('frequency')->options(frequency::cases())`

`enum BudgetFrequency: int 
{
    case Once = 1;
    case Twice = 2;
    case Thrice = 3;
}`

Thanks

**Comment by pixelpeter** [comment]
I have it working like this

```
Forms\Components\Radio::make('status')->required()
    ->label('Status')
    ->options(collect(StatusEnum::cases())->pluck('value', 'name'))
    ->inline(),
```

**Comment by danharrin** [comment]
For selects, you can just pass the class name of the enum into `options(EnumName::class)` AFAIK.

This needs to be done for radio too.

**Reply by pixelpeter** [reply]
I just checked it with my code and this is the result

```php
// this is working
Forms\Components\Select::make('status')
    ->options(StatusEnum::class),

// this is not working
// Expected parameter of type 'array|\Closure|\Illuminate\Contracts\Support\Arrayable', 'string' provided
Forms\Components\Radio::make('status')
    ->options(StatusEnum::class),
```

**Reply by danharrin** [reply]
As I mentioned, this still needs to be added to the Radio component.

**Reply by pixelpeter** [reply]
Ahh. Sorry. I missunderstood your comment

**Reply by danharrin** [reply]
I'll add that functionality to the next release

**Comment by trinketbox** [comment]
Thanks guys, got it working first using similar to pixelpeter suggestion. I’ve yet to try Dan’s, (will be trying that later) but thanks for taking the time. It’s a GREAT package. 🙌
